### ISSUE TITLE
Freezing when fast scanning through audio file (with the mouse)

### ISSUE THREAD

**Issue by dardoor** [mpv-player/mpv#4846]
**EDITED** after gathering more info. The title should be something like: _GUI freezing on progress bar with static video output (on Windows 7)_ **and f.lux** (see the posts below).

### mpv version and platform

v0.26.0, but it started with v0.24.0. mpv 64bit, Windows 7 64bit. (update: it's still here on v.0.27.0)

### Reproduction steps

1. Open a media file with a static image (audio file or picture) **or** a video file that you then pause.
2. Just move the mouse pointer back and forth over the progress bar **or** click and hold to scroll through the file.

### Expected behavior

The pointer should just show the position without freezing the GUI, like in v0.23.0 and earlier.

### Actual behavior

The mpv GUI freezes (twice) for a couple of seconds, after that it works fine. (Audio playback is not interrupted.)

Video recording: [https://webmshare.com/play/6rdMB](https://webmshare.com/play/6rdMB) (also attached: [mpv.zip](https://github.com/mpv-player/mpv/files/1306727/mpv.zip)) You can see how it freezes at 00:20 and at 00:37.

### Log file
(holding LMB and scrolling through)
https://pastebin.com/NWBRymLv

**Comment by haasn** [comment]
Can't reproduce

**Comment by dardoor** [comment]
Some more info:
1. The issue happens even if you just hover over the progress bar with the mouse, you don't have to click anything.
2. The issue is not just for audio files, but anything that has a static video output, even a paused video.
3. You don't have to start hovering over it quickly after opening, even after 20 seconds it will still produce the freezes.

So, an alternative way to reproduce it, with any media file:
1. open media file
2. pause it
3. hover over the progress bar back and forth

It seems the GUI freezes exactly two times.After the two freezes, it continues to work fine.

Here's an example with an audio file:
[https://webmshare.com/play/6rdMB](https://webmshare.com/play/6rdMB)
Notice the two freezes at positions 00:20 and 00:37.

**Comment by haasn** [comment]
Linked page is a 404 for me. Can you reproduce the issue with --no-config, on every file?

**Comment by dardoor** [comment]
Yes, I can. And so far it works on every file I tried, even on a still image file (jpg).

Here's the attached recording, for anyone who can't open that link:
[mpv.zip](https://github.com/mpv-player/mpv/files/1298552/mpv.zip)

**Comment by dardoor** [comment]
Some more info:
Windows 10 on the same PC - no bug.
Windows 10 on the same PC, running in a VM (on W7) - no bug.
Windows 7 on another PC - bug.

So it seems it's somehow Windows 7 related. Or at least it's not reproducible on Windows 10.

**Comment by lachs0r** [comment]
My crystal ball says:

Realtek drivers installed? Uninstall those, they’re buggy.
Creative brand sound card? Buggy drivers.
USB audio? Buggy device.

**Comment by dardoor** [comment]
See my subsequent posts, it has nothing to do with audio. I edited the first post, hopefully it's clearer this way. Can't edit the title, though.
What triggers the issue is a **static video output/picture** (or the lack of it, as typical for audio files). And so far I only saw it on Windows 7, not 10.

**Comment by haasn** [comment]
@dardoor From your recording it seems as though your mouse cursor is freezing as well. Is that an issue with the recording or does your mouse cursor actually lock up?

**Comment by dardoor** [comment]
My mouse cursor locks up too. Actually, the entire screen locks up. For example, if I have something playing somewhere (say, a video in VLC) it will freeze too.

I added some more info here: [#4921](https://github.com/mpv-player/mpv/issues/4921).
Just like that one, this issue is caused by **f.lux** or similar programs (SunsetScreen), running on Windows 7 and with newer versions of mpv.
Just like there, if I exit Flux this freezing discussed here (with a static image) doesn't happen anymore.

**Comment by haasn** [comment]
If your entire screen locks up then it's a bug in your graphics driver / compositor one way or the other. A single application could not cause your entire screen to lock up. And the fact that nobody else seems to be able to replicate it, and it goes away on different windows versions (with the same mpv version) leads me to suspect that it's not an mpv bug.

If it's also a regression in mpv then it may be possible to work around it somehow, but I think the chances of that are slim.

**Comment by dardoor** [comment]
I did some more testing and I can make the bug go away with **vo=direct3d**. Does mpv use opengl by default on Windows?
I skimmed through the v.0.24 changelog and there are a bunch of opengl changes, so I wonder if that's related and/or if there's a better workaround.

Anyway, I'm ok with this workaround, so I won't bother you with this any longer.
But in case it affects more people, it might be worth looking into. As I said, I can easily reproduce this on two different PCs with W7 and Flux... Although both PCs have an AMD Radeon card (an old 4xxx and a 7xxx), so that might be related as well.
